js/main.js

```diff
@@ -0,0 +1,156 @@
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+
+function redraw(){
+
+  var height = canvas.getBoundingClientRect().height;
+  var width = canvas.getBoundingClientRect().width;
+
+  //console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight 
+ * @param {number} canvasWidth 
+ * @returns {Object}
+ */
+function createGameBoard(canvasHeight, canvasWidth){
+  var gameBoard = {
+    startPointX: 0,
+    startPointY: 0,
+    gameHeight: 10,
+    gameWidth: 10,
+    board: []
+  }
+
+  var heightScaler = canvasHeight / gameBoard.gameHeight;
+  var widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  if (widthScaler > heightScaler) {
+    widthScaler = heightScaler;
+  } else {
+    heightScaler = widthScaler;
+  }
+
+  var middlePoint = canvasWidth / 2 - (widthScaler * gameBoard.gameWidth/2);
+  gameBoard.startPointX = middlePoint;
+
+  currX = gameBoard.startPointX;
+  currY = gameBoard.startPointY;
+
+  for(var i = 0 ; i < gameBoard.gameHeight ; ++i){
+    var rowOfSquares = [];
+    for(var j = 0 ; j < gameBoard.gameWidth ; ++j){
+
+      var square = {
+        xCoord: currX,
+        yCoord: currY,
+        width: widthScaler,
+        height: heightScaler,
+        character: '-'
+      };
+
+      rowOfSquares.push(square);
+
+      currX += widthScaler;
+      
+    } 
+    gameBoard.board.push(rowOfSquares);
+    currX = gameBoard.startPointX;
+    currY += heightScaler;
+  }
+
+  return gameBoard;
+}
+
+/**
+ * Converts ranNumber from a float to a character in the
+ * english alphabet based on what interval it is within the charFrequency. 
+ * @param {array} charFrequency an array of floats where element at index i 
+ *    less element index i - 1 is the frequency that the i'th letter occurs in 
+ *    the english language
+ * @param {number} ranNumber a random float between 0-101.432 
+ * @return {String} 
+ */
+function getCharater(charFrequency, ranNumber){
+  var index = 0;
+
+  for(var i = 0 ; i < 26 ; ++i){
+    if (ranNumber < charFrequency[i]){
+      index = i;
+      break;
+    }
+  }
+
+  return String.fromCharCode(97 + index)
+}
+
+/**
+ * Based on a weighted random distribution, it populates the wordSearchBoard
+ * with random letters.
+ * @param {object} wordSearchBoard 
+ */
+function randomizeCharacters(wordSearchBoard) {
+
+  /** @const {number} */ 
+  const frequency = [8.498, 9.989, 12.191, 16.445, 27.607, 29.835, 31.85, 37.944,
```

Comment:
Have you looked into running a formatter over this code?

A lot of Google projects enforce that all PRs are formatted before being submitted.

The formatter Google uses for JS is described [here](http://probst.io/blog/2015/01/15/clang-format-for-javascript/#:~:text=clang%2Dformat%20for%20JavaScript&text=clang%2Dformat%20is%20a%20very,manually%20nor%20use%20a%20linter). It seems like you should be able to run it externally. If that doesn't work, I'd still recommend finding a different open source autoformatter. It's very nice to not have to worry about indentation when you code and just let the formatter handle it for you.

Reply:
Updated to be properly formatted.